ee/control/control_test.go

```diff
@@ -0,0 +1,163 @@
+package control
+
+import (
+	"io"
+	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type (
+	mockConsumer struct {
+		updateFn func()
+	}
+	mockSubscriber struct {
+		pingFn func()
+	}
+)
+
+func (mc *mockConsumer) Update(io.Reader) {
+	if mc.updateFn != nil {
+		mc.updateFn()
+	}
+}
+func (mc *mockSubscriber) Ping() {
+	if mc.pingFn != nil {
+		mc.pingFn()
+	}
+}
+
+type nopDataProvider struct{}
+
+func (dp nopDataProvider) Get(subsystem, cachedETag string) (etag string, data io.Reader, err error) {
+	return "", nil, nil
+}
+
+func testCS() *ControlService {
+	data := nopDataProvider{}
+	cs := &ControlService{
```

Comment:
Feels a little weird. I think a lot of this would vanish if you called `New()`, might be reduceable to `New(data)`. But I don't know if you have plans to expand the complexity here. 

I guess even if you keep `testCS` I think you should use New and not instantiate directly.